federation/pkg/kubefed/init/init.go

```diff
@@ -583,9 +587,51 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 		},
 	}
 
+	if dnsProviderConfig != "" {
+		// Create a dns-provider secret from the provided config file
+		dnsProviderConfigBytes, err := ioutil.ReadFile(dnsProviderConfig)
+		if err != nil {
+			return nil, err
+		}
+		secret := &api.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dnsProviderSecretName,
+				Namespace: namespace,
+			},
+			Data: map[string][]byte{
+				dnsProviderSecretKey: dnsProviderConfigBytes,
+			},
+		}
+		secret, err = clientset.Core().Secrets(namespace).Create(secret)
```

Comment:
The current mechanism to cleanup is to do something like delete the federation-system namespace and start over again.